influence_simulator/utils.py

```diff
@@ -0,0 +1,86 @@
+import resource
+from contextlib import contextmanager
+from datetime import timedelta
+from logging import getLogger
+from time import perf_counter
+from typing import Callable
+
+import networkx as nx
+from lightning_diffusion.models.diffusion_model import DiffusionModel
+
+
+logger = getLogger("influence-simulator")
+
+
+def format_delta(seconds: float, digits: int = -1) -> str:
```

Comment:
this is a time of execution, not a time as a simulation step / epoch, right?

Reply:
It's just a small utility function to format `perf_counter` output to a more readable format. It will be especially useful in WCSS, as there we need an upper bound for our processing time